subsys/bootloader/cmake/sign.cmake

```diff
@@ -196,3 +196,26 @@ foreach (slot ${slots})
     )
 
 endforeach()
+
+if (CONFIG_BUILD_S1_VARIANT AND NOT CONFIG_BOOTLOADER_MCUBOOT)
+  # B0 will boot the app directly, create the DFU zip. If MCUBoot is enabled,
+  # the DFU zip files is generated by the MCUBoot build code.
+
+  include(${NRF_DIR}/cmake/fw_zip.cmake)
+
+  # Generate zip file with both update candidates
+  set(s0_name signed_by_b0_s0_image.bin)
+  set(s0_bin_path ${PROJECT_BINARY_DIR}/${s0_name})
+  set(s1_name signed_by_b0_s1_image.bin)
+  set(s1_bin_path ${PROJECT_BINARY_DIR}/${s1_name})
+
+  generate_dfu_zip(
+    ${PROJECT_BINARY_DIR}/dfu_application.zip # Zip out
+    "${s0_bin_path};${s1_bin_path}"
+    "s0_image_signed_kernel_hex_target;s1_image_signed_kernel_hex_target" # dependencies
+    "type=application"
+    "${s0_name}load_address=$<TARGET_PROPERTY:partition_manager,PM_S0_ADDRESS>"
+    "${s1_name}load_address=$<TARGET_PROPERTY:partition_manager,PM_S1_ADDRESS>"
+    "version_B0=${CONFIG_FW_INFO_FIRMWARE_VERSION}"
+    )
```

Comment:
Not sure I understand the value of a function for this, as you anyway need know all the args being passed to the python script here.

Have you tried something like this, and if yes, but didn't work, maybe you can explain.
```suggestion
 add_custom_command(
   TARGET <signed hex file target ?>
   POST_BUILD
    COMMAND
    ${PYTHON_EXECUTABLE}
    ${NRF_DIR}/scripts/bootloader/generate_zip.py
    --bin-files "${s0_bin_path};${s1_bin_path}"
    --output ${PROJECT_BINARY_DIR}/dfu_application.zip
    "board=${CONFIG_BOARD}"
    "soc=${CONFIG_SOC}"
    "type=application"
    "${s0_name}load_address=$<TARGET_PROPERTY:partition_manager,PM_S0_ADDRESS>"
    "${s1_name}load_address=$<TARGET_PROPERTY:partition_manager,PM_S1_ADDRESS>"
    "version_B0=${CONFIG_FW_INFO_FIRMWARE_VERSION}"
    BYPRODUCTS ${PROJECT_BINARY_DIR}/dfu_application.zip
    )
```

Reply:
Your proposal use more code lines (15 vs 8) and this function is called 4 times (60 vs 32 total), so that is enough motivation IMO. In addition there are several lines which are duplicated (soc, board, version) which makes it harder to maintain, and easier to make mistakes.

Reply:
Not sure how you count, but I get 15 vs 12, so that's only 3 more lines 😉 , meaning 12 lines in total, and we save an additional function of 25 lines (without blank lines and `)`).

So not such a huge difference (which makes sense, as the CMake function is almost a one-to-one match of the Python script API).

But I will say that after the arguments are named, the code is much more readable, so will not block the PR over this.

```
generate_dfu_zip(
  OUTPUT
  ${PROJECT_BINARY_DIR}/dfu_application.zip
  BIN_FILES
  ${PROJECT_BINARY_DIR}/app_update.bin
  DEPENDENCIES
  mcuboot_sign_target
  TYPE
  application
  SCRIPT_PARAMS
  "load_address=$<TARGET_PROPERTY:partition_manager,PM_APP_ADDRESS>"
  "version_MCUBOOT=${CONFIG_MCUBOOT_IMAGE_VERSION}"
)
```

Reply:
I count with a bias ;)

Reply:
> I count with a bias ;)

I must say you do, cause I only find 3 places where you call this function, and not 4 :stuck_out_tongue: